docs/docs/development/hardware-integration/pinctrl.mdx

```diff
@@ -0,0 +1,336 @@
+:::
+
+## Boards, Shields, and Modules
+
+:::tip
+Some boards, such as the Seeed Studio Xiao, already have some of their pins configured for particular communication protocols. If this is the case, then you can use the pre-configured bus rather than configuring one manually (you merely need to identify the correct label for the bus node).
+:::
+
+Pin control is always defined for a _board_, never for a shield:
+
+- If you are defining a keyboard that is a _board_, then you will be editing and adding on to the files in your keyboard's folder as usual.
+- If your keyboard consists of a _board and a shield_, then you should define a new folder inside of your shield's folder called `boards`. You will need to add a `<board>.overlay` for each board to be used with said shield. Take for example the file structure of the "sofle" shield:
+  ```plaintext
+  boards/shields/sofle/
+  ├── boards/
+  │   ├── nice_nano.overlay
+  │   ├── nice_nano_v2.overlay
+  │   ├── nrfmicro_11.overlay
+  │   └── nrfmicro_13.overlay
+  └── <sofle shield defining files>
+  ```
+  Note that you will need to define a separate overlay _for each_ of the boards to be used with the shield.
+
+:::info
+Assume that the shield that you are using is found in-tree of ZMK or within an external module, and _does not_ contain the overlay for the board that you wish to use.
+If this is the case, then you should fork the source repository and add the overlay to the fork. Use said fork to build your firmware, and potentially submit a PR to upstream.
+:::
+
+## Predefined Nodes
+
+Many boards will already have some pins configured for particular protocols. We recommend using these pins whenever possible.
+
+<InterconnectTabs items={Metadata} gpio={false} />
+
+Note that some boards may have these pins configured to fit the above pinouts, rather than the capabilities of the MCU. Most notably, boards using the nRF52840 often end up putting low-frequency pins at the locations of the SPI buses above. More on low-frequency pins in the next section.
+
+## Pin Selection
+
+Prior to setting up pin control, you will want to make sure that the pins you use are suitable for the driver that you are using. For this purpose, you will most likely need to consult the datasheet of your MCU/SoC directly. Different MCUs/SoCs have different restrictions in regards to this. The nRF52840 and RP2040 are excellent examples:
+
+- The nRF52840 has its pins marked as either low frequency or high frequency.
+  - Low frequency pins are limited to a frequency of 10KHz, making them unusable for I2C, UART, and (most of) SPI.
```

Comment:
```suggestion
  - Low frequency pins should not be used with signals with frequencies exceeding 10KHz, so they are not ideal for I2C, UART, and (most of) SPI.
```
I think it'd be better to document what happens when you do use low frequency pins (interference with radio) rather than say don't use them, given that many designs seem to be using them anyway. Would you agree?